modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -811,6 +1280,67 @@ private void sendGenerateEncryptionKeyRequest(GenerateEncryptionKeyFuture fut) t
         ctx.io().sendToGridTopic(rndNode.id(), TOPIC_GEN_ENC_KEY, req, SYSTEM_POOL);
     }
 
+    /**
+     * @param grpIds Cache group IDs.
+     * @param skipDirty Dirty page skip flag.
+     * @throws IgniteCheckedException If failed.
+     */
+    private void startReencryption(Collection<Integer> grpIds, boolean skipDirty) throws IgniteCheckedException {
```

Comment:
Could you divide the class into subclasses and use a compositional approach?

Reply:
I do not quite understand the essence of the problem with this particular method and what kind of decomposition needs to be done with it.